File: spec/java/src/io/kaitai/struct/spec/TestOpaqueExternalType02Parent.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.OpaqueExternalType02Parent;
import io.kaitai.struct.testformats.OpaqueExternalType02Child;
import org.testng.annotations.Test;
import static org.testng.Assert.*;
public class TestOpaqueExternalType02Parent extends CommonSpec {

    @Test
    public void testOpaqueExternalType02Parent() throws Exception {
        OpaqueExternalType02Parent r = OpaqueExternalType02Parent.fromFile(SRC_DIR + "term_strz.bin");

        assertEquals(r.parent().child().s1(), "foo");
        assertEquals(r.parent().child().s2(), "bar");
        assertEquals(r.parent().child().s3().s3(), "|baz@");
    }
}

File: spec/python/spec/test_opaque_external_type_02_parent.py
import unittest
from testformats.opaque_external_type_02_parent import OpaqueExternalType02Parent

class TestOpaqueExternalType02Parent(unittest.TestCase):
    def test_opaque_external_type_02_parent(self):
        with OpaqueExternalType02Parent.from_file('src/term_strz.bin') as r:
            self.assertEqual(r.parent.child.s1, "foo")
            self.assertEqual(r.parent.child.s2, "bar")
            self.assertEqual(r.parent.child.s3.s3, "|baz@")
